### PR TITLE
Use Set to detect banned teapot segments

### PR DESCRIPTION
**Pull request by jc00ke**
The previous implementation was `O(n^2)` where this implementation using a `Set` is `O(n)`.

### CODE REVIEW

File: app/middlewares/rack/teapot.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+require "set"
```
Comment (astronaut-wannabe) [inline app/middlewares/rack/teapot.rb]: rubocop stuff
```suggestion
require 'set'
```

File: app/middlewares/rack/teapot.rb [hunk 2]
```diff
@@ -30,13 +31,8 @@ def call env
...
 
-      # check if the requested path piecies
-      path.split('/').each do |path_piece|
-        PATHS.include? path_piece
-
-        # send a 418 code, if it looks like it’s a pen test kind of request
-        return i_am_a_teapot if PATHS.include? path_piece
-      end
+      # send a 418 code, if it looks like it’s a pen test kind of request
+      return i_am_a_teapot if path.split("/").any? { |segment| BANNED_SEGMENTS.member? segment }
```
Comment (astronaut-wannabe) [inline app/middlewares/rack/teapot.rb]: rubocop stuff
```suggestion
      return i_am_a_teapot if path.split('/').any? { |segment| BANNED_SEGMENTS.member? segment }
```

**Comment by veganstraightedge** [comment]
Thanks @jc00ke! As @astronaut-wannabe pointed out, there are a couple Rubocop little changes needed, otherwise this looks great. 🎉

**Comment by veganstraightedge** [comment]
Awesome.

**Pull request merged**